addons/sale_coupon/models/coupon_program.py

```diff
@@ -12,19 +11,19 @@ class CouponProgram(models.Model):
 
     # The api.depends is handled in `def modified` of `sale_coupon/models/sale_order.py`
     def _compute_order_count(self):
-        program_orders = self.env["sale.order.line"].read_group(
+        program_orders = self.env['sale.order.line'].read_group(
             domain=[
-                ("product_id", "in", self.mapped("discount_line_product_id").ids),
+                ('product_id', 'in', self.discount_line_product_id.ids),
             ],
-            fields=["order_id"],
-            groupby=["product_id", "order_id"],
-            lazy=False,
+            fields=['order_id:count_distinct'],
+            groupby=['product_id'],
         )
-        mapped_data = defaultdict(set)
-        for value in program_orders:
-            mapped_data[value["product_id"][0]] |= {value["order_id"][0]}
+        mapped_data = {
+            value['product_id'][0]: value['product_id_count']
```

Comment:
Shouldn't we store "order_id" instead of "product_id_count"? After some testing,  now it is displaying the number of products on the sales orders that had the promotion discount applied to, rather than the number of orders.

Reply:
@Feyensv it seems to count correctly now with brief testing. Are the commits ready to be squashed now?

Reply:
Yeah, feel free to, that's what fixup commits are for (little autosquash rebase and they magically merge with the targeted commit)

Reply:
Oh I did not know that. That is so useful to know!